src/index.js

```diff
@@ -33,6 +33,12 @@ export default function <ResultFn: (...Array<any>) => mixed>(resultFn: ResultFn,
     return lastResult;
   };
 
+  Object.defineProperty(result, 'length', {
```

Comment:
could you just do `result.length = resultFn.length`?

Reply:
No, function.length is readonly property, reflecting the numbers of arguments. The only way to override it - define a descriptor.
Actually, the best way here is to pass through all the descriptors from the original function, including name, and any static prop, that may exist, but I decide to keep your code compact.